src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3802,6 +3813,86 @@ GenTree* Lowering::TryLowerAndOpToResetLowestSetBit(GenTreeOp* andNode)
     return blsrNode;
 }
 
+//----------------------------------------------------------------------------------------------
+// Lowering::TryLowerAndOpToAndNot: Lowers a tree AND(X, NOT(Y)) to HWIntrinsic::AndNot
+//
+// Arguments:
+//    andNode - GT_AND node of integral type
+//
+// Return Value:
+//    Returns the replacement node if one is created else nullptr indicating no replacement
+//
+// Notes:
+//    Performs containment checks on the replacement node if one is created
+GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
+{
+    assert(andNode->OperIs(GT_AND) && varTypeIsIntegral(andNode));
+
+    GenTree* opNode  = nullptr;
+    GenTree* notNode = nullptr;
+    if (andNode->gtGetOp1()->OperIs(GT_NOT))
+    {
+        notNode = andNode->gtGetOp1();
+        opNode  = andNode->gtGetOp2();
+    }
+    else if (andNode->gtGetOp2()->OperIs(GT_NOT))
+    {
+        notNode = andNode->gtGetOp2();
+        opNode  = andNode->gtGetOp1();
+    }
+
+    if (opNode == nullptr)
+    {
+        return nullptr;
+    }
+
+    // We will avoid using "andn" when one of the operands is (likely to be) in memory:
+    // "and"'s RMW form provides for a shorter instruction sequence in that case.
```

Comment:
Comment should likely be updated. This isn't covering any memory operand but rather specifically the case where an operand is both the source and destination and therefore it removes an otherwise necessary memory access that would make `andn` a worse choice.

Reply:
Sure? suggestions?

Reply:
```suggestion
    // We want to avoid using "andn" when one of the operands is a source and the destination and is coming from memory
    // In this scenario, we will get smaller and likely faster code by using the RMW encoding of `and`
```

Reply:
Done.